File: interface/python_binding/pyktx/ktx_pack_bcn_quality_levels.py
# Copyright (c) 2023, Shukant Pal and Contributors
# SPDX-License-Identifier: Apache-2.0

from enum import IntEnum


class KtxPackBCnQualityLevels(IntEnum):
    """
    Options specifiying high-level BC1, BC3, and BC7 encoding quality levels

    These enums will be mapped to their corresponding BC1/BC3 or BC7 quality
    values. For BC1/BC3, these are mapped to [0, 19]. For BC7, these are mapped
    to particular bitsets of internal flags.
    """

    FASTEST = 0
    """Fastest compression. For BC1/BC2/BC3, this maps to 0."""

    FASTER = 1
    """Faster compression. For BC1/BC3, this maps to 2."""

    FAST = 2
    """Fast compression. For BC1/BC3, this maps to 5."""

    MEDIUM = 3
    """Medium compression. For BC1/BC3, this maps to 10."""

    THOROUGH = 4
    """Thorough compression. For BC1/BC3, this maps to 15."""

    EXHAUSTIVE = 5
    """Exhaustive compression. For BC1/BC3, this maps to 19."""

    MAX = EXHAUSTIVE
    """Maximum supported quality level."""

File: interface/java_binding/src/main/java/org/khronos/ktx/KtxPackBCnQualityLevels.java
/*
 * Copyright (c) 2026, Khronos Group and Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

package org.khronos.ktx;

/**
 * Options specifying BC1, BC2, BC3, and BC7 encoding quality levels.<br>
 * <br>
 * These constants can be passed to
 * {@link KtxBCnParams#setBCnCompressionQuality(int)}.
 */
public class KtxPackBCnQualityLevels {

  /**
   * Fastest compression. For BC1/BC2/BC3, this maps to 0.
   */
  public static final int FASTEST = 0;

  /**
   * Faster compression. For BC1/BC2/BC3, this maps to 2.
   */
  public static final int FASTER = 1;

  /**
   * Fast compression. For BC1/BC2/BC3, this maps to 5.
   */
  public static final int FAST = 2;

  /**
   * Medium compression. For BC1/BC2/BC3, this maps to 10.
   */
  public static final int MEDIUM = 3;

  /**
   * Thorough compression. For BC1/BC2/BC3, this maps to 15.
   */
  public static final int THOROUGH = 4;

  /**
   * Exhaustive compression. For BC1/BC2/BC3, this maps to 19.
   */
  public static final int EXHAUSTIVE = 5;

  /**
   * Returns a string representation of the given quality level
   *
   * @param n The quality level
   * @return A string representation of the given quality level
   */
  public static String stringFor(int n) {
    switch (n) {
    case FASTEST: return "FASTEST";
    case FASTER: return "FASTER";
    case FAST: return "FAST";
    case MEDIUM: return "MEDIUM";
    case THOROUGH: return "THOROUGH";
    case EXHAUSTIVE: return "EXHAUSTIVE";
    }
    return "[Unknown KtxPackBCnQualityLevel]";
  }

  /**
   * Private constructor to prevent instantiation
   */
  private KtxPackBCnQualityLevels() {
    // Prevent instantiation
  }

}
